db/cryptodata/decrypter.go

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cryptodata
+
+import (
+	"encoding/json"
+	"strings"
+	"encoding/base64"
+)
+
+// DecryptFunc is function that decrypts arbitrary data
+type DecryptFunc func(inData []byte) (data [] byte, err error)
+
+// CheckEncryptedFunc is used to check if JSON data is encrypted
+type CheckEncryptedFunc func(inData []byte) (isEncrypted bool)
+
+// EncryptionCheck is used to check for data to contain encrypted marker
+type EncryptionCheck struct {
+	// IsEncrypted returns true if data was marked as encrypted
+	IsEncrypted bool `json:"encrypted"`
+}
+
+// ArbitraryDecrypter is interface for decrypting groups of data
```

Comment:
Should be `ArbitraryDecrypter represents decrypter that looks for encrypted values inside arbitrary data and returns the data with the values decrypted`